### PR TITLE
converting to command reference

### PR DESCRIPTION
**Pull request by flossypurse**
This PR adds a Command reference to the docs.

Preview: https://deploy-preview-722--mystifying-fermi-1bc096.netlify.app/docs/reference/commands/

### CODE REVIEW

File: docs/reference/commands.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+---
+id: commands
+title: Command reference
+description: A Command is a requested action issued by a Worker to the Temporal Cluster, after the Worker has finished completing the execution of a Workflow Task.
```
Comment (mfateev) [inline docs/reference/commands.md:4]: "finished completing the execution" is hard to parse for me.

**Comment by Sushisource** [comment]
See also https://github.com/temporalio/api/pull/121 which is taking a while to merge but adds proto doc comments for commands as well. Agree with @sw-yx we should probably link to those from this (which will be useful after that PR is merged)

**Comment by Sushisource** [comment]
Also I think the `Description:` preamble on every first bullet is unneeded

File: docs/reference/commands.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import * as WhatIsAnEvent from '../content/what-is-an-event.md'
+
+### What is a Command?
+
+A Command is a requested action issued by a <preview page={WhatIsAWorker}>Worker</preview> to the <preview page={WhatIsATemporalCluster}>Temporal Cluster</preview> after the Worker has finished completing the execution of a <preview page={WhatIsAWorkflowTask}>Workflow Task</preview>.
+
+The action that the Cluster takes is recorded in the <preview page={WhatIsAWorkflowExecution}>Workflow Execution's</preview> <preview page={WhatIsAnEventHistory}>Event History</preview> as an <preview page={WhatIsAnEvent}>Event</preview>.
+
+During the execution of a Workflow Task there may be several Commands that are generated.
+The Commands are batched and sent to the Cluster as part of the Workflow Task completion request, after the Workflow Task has progressed as far as it can.
```
Comment (mfateev) [inline docs/reference/commands.md:25]: I think we need to explain that commands are recorded as events and in the history we always see
WorkflowTaskStarted
WorkflowTaskComleted
<command event 1>
<command event 2>

File: docs/reference/commands.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+### CompleteWorkflowExecution
+
+- Description: This Command is triggered when the last Workflow Task of the Workflow Execution has finished executing successfully.
+  Often this is when the last line of code in a Workflow Definition has been execution.
+- Awaitable: No, a Workflow Execution can not await on the action resulting from this Command.
+- Corresponding Event: [WorkflowExecutionComplete](/docs/reference/events/#workflowexecutioncompleted)
```
Comment (mfateev) [inline docs/reference/commands.md:32]: WorkflowExecutionComplete**d**

File: docs/reference/commands.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+The action that the Cluster takes is recorded in the <preview page={WhatIsAWorkflowExecution}>Workflow Execution's</preview> <preview page={WhatIsAnEventHistory}>Event History</preview> as an <preview page={WhatIsAnEvent}>Event</preview>.
+
+During the execution of a Workflow Task there may be several Commands that are generated.
+The Commands are batched and sent to the Cluster as part of the Workflow Task completion request, after the Workflow Task has progressed as far as it can.
+
+### CompleteWorkflowExecution
+
+- Description: This Command is triggered when the last Workflow Task of the Workflow Execution has finished executing successfully.
+  Often this is when the last line of code in a Workflow Definition has been execution.
```
Comment (mfateev) [inline docs/reference/commands.md:30]: "when the last Workflow Task of the Workflow Execution has finished executing successfully." is confusing.

This command indicates to the service that workflow execution is completed. This is always the last command in the workflow execution history. 

We can also mention that this command contains the result of a workflow execution.

File: docs/reference/commands.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+- Awaitable: No, a Workflow Execution can not await on the action resulting from this Command.
+- Corresponding Event: [WorkflowExecutionComplete](/docs/reference/events/#workflowexecutioncompleted)
+
+### ContinueAsNewWorkflowExecution
+
+- Description: This Command is triggered when there is a call to ContinueAsNew from within the Workflow.
```
Comment (mfateev) [inline docs/reference/commands.md:36]: Should we explain the semantic of contiuneAsNew here? Otherwise we should link to the corresponding doc.
I would also mention that this is the last command in the workflow execution history.

File: docs/reference/commands.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+### FailWorkflowExecution
+
+- Description: This Command is triggered when the Workflow Execution returns an error or an exception is thrown.
+- Awaitable: No, a Workflow Execution can not await on the action resulting from this Command.
+- Corresponding Event: [WorkflowExecutionFailed](/docs/reference/events/#workflowexecutionfailed)
+
+### CancelWorkflowExecution
+
+- Description: This Command is triggered when the Workflow Execution has accepted the Cancellation Request.
```
Comment (mfateev) [inline docs/reference/commands.md:48]: This description is not correct. 

This command is triggered when workflow successfully completed cleanup after receiving a cancellation request (which had to be present in the history as WorkflowExecutionCancelRequestedEvent). This is the last command in the workflow history.

File: docs/reference/commands.md (hunk 1 above)
Comment (flossypurse) [reply docs/reference/commands.md:4]: reworded

File: docs/reference/commands.md (hunk 5 above)
Comment (flossypurse) [reply docs/reference/commands.md:36]: Going to define that in a separate page and link when it's up.

**Review by sw-yx — APPROVED** [review]
